pkg/component/extensions/operatingsystemconfig/operatingsystemconfig_test.go

```diff
@@ -322,6 +322,11 @@ var _ = Describe("OperatingSystemConfig", func() {
 
 		Describe("#Deploy", func() {
 			It("should successfully deploy the shoot access secret for the cloud config downloader", func() {
+				defer test.WithVars(
+					&DownloaderConfigFn, downloaderConfigFn,
+					&OriginalConfigFn, originalConfigFn,
+				)()
```

Comment:
Why is this necessary? I don't see other changes in this file 👀

Reply:
We need it because otherwise it will call the original `OriginalConfigFn` and `DownloaderConfigFn` and this will panic with reason that feature gate `ContainerdRegistryHostsDir` is not registered.
In this test we verify that the shoot access secret gets created. Hence, I don't think we need to call the original `OriginalConfigFn` and `DownloaderConfigFn` funcs. As far as I see we don't test/verify something that comes from `OriginalConfigFn` and `DownloaderConfigFn`.